### PR TITLE
Add CPU time and MemCopy to the timeline.

### PR DESCRIPTION
**Pull request by panyx0718**

### CODE REVIEW

File: paddle/fluid/platform/device_tracer.cc [hunk 1]
```diff
@@ -54,6 +55,36 @@ uint64_t kAlignSize = 8;
     }                                                                      \
   } while (0)
 
+std::string MemcpyKind(CUpti_ActivityMemcpyKind kind) {
+  switch (kind) {
+    case CUPTI_ACTIVITY_MEMCPY_KIND_HTOD:
+      return "MEMCPYHtoD";
```
Comment (jacquesqiao) [inline paddle/fluid/platform/device_tracer.cc]: Maybe just `HtoD` is ok? There are many duplicates `MEMCPY`

Comment (panyx0718) [reply paddle/fluid/platform/device_tracer.cc]: Done

Comment (panyx0718) [reply paddle/fluid/platform/device_tracer.cc]: I Added it back. User can search MEMCPY to find all copy.

File: paddle/fluid/platform/profiler.proto [hunk 2]
```diff
@@ -15,12 +15,17 @@ limitations under the License. */
 syntax = "proto2";
 package paddle.platform.proto;
 
+message MemCopy { optional uint64 bytes = 3; }
```
Comment (jacquesqiao) [inline paddle/fluid/platform/profiler.proto]: why start from 3?

File: paddle/fluid/platform/device_tracer.cc [hunk 3]
```diff
@@ -197,7 +260,24 @@ class DeviceTracerImpl : public DeviceTracer {
...
-      event_times[event->name()].push_back(r.end_ns - r.start_ns);
+    }
+
+    for (const CPURecord &r : cpu_records_) {
+      auto *event = profile_pb.add_events();
+      event->set_name(r.name);
+      event->set_start_ns(r.start_ns);
+      event->set_end_ns(r.end_ns);
+      event->set_stream_id(r.thread_id);
+      event->set_device_id(-1);
```
Comment (jacquesqiao) [inline paddle/fluid/platform/device_tracer.cc]: use a const value to represent CPU instead of `-1`, like
```cpp
const kCpuDeviceId = -1;
```

**Pull request merged**